### PR TITLE
Add new buttons to the web connector

### PR DESCRIPTION
**Pull request by bakic**
L'ajout des boutons suivants au connecteur web:

webQuickReply
webPostbackButton
webUrlButton
Création de nouvelles classes:

WebCard
WebCarousel

### CODE REVIEW

File: bot/connector-web/src/main/kotlin/WebBuilders.kt [hunk 1]
```diff
@@ -67,18 +75,87 @@ fun <T : Bus<T>> T.webButton(
...
+    subTitle: CharSequence,
+    buttons: List<Button>
+): WebCard = WebCard(
+    title = title,
+    subTitle = subTitle,
+    buttons = buttons
+)
+
+fun <T : Bus<T>> T.webCardWithAttachement(
+    title: CharSequence,
```
Comment (elebescond) [inline bot/connector-web/src/main/kotlin/WebBuilders.kt]: title and subtitle mandatory? Sometine we only want an image + 1 button ?

Comment (bakic) [reply bot/connector-web/src/main/kotlin/WebBuilders.kt]: Fixed

File: bot/connector-web/src/main/kotlin/send/ButtonType.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.tock.bot.connector.web.send
+
+enum class ButtonType {
+    web_url, postback, quick_reply, quick_reply_url, none
```
[the remark this answers is not in the document]
Comment (bakic) [reply bot/connector-web/src/main/kotlin/send/ButtonType.kt]: Fixed

File: bot/connector-web/src/main/kotlin/send/MediaCarousel.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package ai.tock.bot.connector.web.send
+
+import ai.tock.bot.connector.media.MediaMessage
+import ai.tock.bot.engine.message.GenericElement
+import ai.tock.bot.engine.message.GenericMessage
+
+/**
+ * A list of [WebCard].
+ */
+data class WebCarousel(val cards: List<WebCard>) : MediaMessage {
```
[the remark this answers is not in the document]
Comment (bakic) [reply bot/connector-web/src/main/kotlin/send/MediaCarousel.kt]: Fixed

File: bot/connector-web/src/main/kotlin/send/WebButton.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * limitations under the License.
+ */
+
+package ai.tock.bot.connector.web.send
+
+import ai.tock.bot.engine.action.SendChoice
+import ai.tock.bot.engine.message.Choice
+import ai.tock.shared.mapNotNullValues
+
+data class WebButton(
```
[the remark this answers is not in the document]
Comment (bakic) [reply bot/connector-web/src/main/kotlin/send/WebButton.kt]: The WebButton class was left to ensure backward compatibility. This class should disappear in the next versions.
I will mark it as deprecated

**Comment by vsct-jburet** [comment]
merged

**Pull request closed without merging**